cmd/virt-handler/virt-handler.go

```diff
@@ -279,14 +280,17 @@ func (app *virtHandlerApp) Run() {
 
 	go vmController.Run(10, stop)
 
-	logger.V(1).Infof("metrics: max concurrent requests=%d", app.MaxRequestsInFlight)
-	http.Handle("/metrics", promvm.Handler(app.MaxRequestsInFlight))
+	errCh := make(chan error)
+	go app.runPrometheusServer(errCh, certStore)
 
-	err = http.ListenAndServeTLS(app.ServiceListen.Address(), certStore.CurrentPath(), certStore.CurrentPath(), nil)
-	if err != nil {
-		log.Log.Reason(err).Error("Serving prometheus failed.")
-		panic(err)
-	}
+	// wait for server to exit
+	<-errCh
+}
+
+func (app *virtHandlerApp) runPrometheusServer(errCh chan error, certStore certificate.FileStore) {
```

Comment:
do we need to actually return the certStore?

Reply:
Thanks @fromanirh :)
Note that this function doesn't return anything, it only gets the certStore as an argument

Reply:
ofc right, I just misread the code.